pyuploadcare/api_resources.py

```diff
@@ -95,6 +115,32 @@ def _api_uri(self):
     def _api_storage_uri(self):
         return 'files/{0}/storage/'.format(self.uuid)
 
+    def generate_query_for_authenticated_url(self, path=None):
+        """Generate query for token authenticated URLs (signed URLs) if feature is on.
+
+        Implementation depends on cdn_provider.
+
+        For more information please refer to docs https://uploadcare.com/docs/delivery/file_api/#authenticated-urls
+        """
+        if conf.cdn_provider is None:
```

Comment:
Проверку на необходимость подписми вынести в cdn_url

Reply:
semantically, `cdn_provider` shouldn't affect if we need to generate signature or not. It can be used for other orthogonal features as well.
I suggest to add one more setting, e.g. `conf.sign_urls = False`
(and `conf.sign_uploads = False` too :) )